src/main/java/addon/Task.java

```diff
@@ -0,0 +1,132 @@
+package addon;
+
+import java.time.LocalDate;
+
+/**
+ * Parent class Task, contains doneness and task name
+ */
+public class Task {
+
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Task constructor.
+     * @param description Name of task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public String toString() {
+        return ((isDone ? "[X] " : "[ ] ") + this.description);
+    }
+
+    /**
+     * Checks if the description contains the keyword provided.
+     * @param str String to be queried.
+     * @return true if found, false otherwise.
+     */
+    public boolean nameContains(String str) {
```

Comment:
Alternatively, maybe the method name "doesNameContain" seems more appropriate for a boolean?